compiler/crates/relay-compiler/src/config.rs

```diff
@@ -843,6 +851,9 @@ struct ConfigFileProject {
 
     #[serde(default)]
     js_module_format: JsModuleFormat,
+
+    #[serde(default)]
+    pub node_interface_id_field: Option<StringKey>,
```

Comment:
can we add comment here, and also to the compiler README.md that explains the purpose of this field

Reply:
Also, I think this is something from the internal discussion about this PR.

I think we should add this field to` FeatureFlags` here:
https://github.com/facebook/relay/blob/59c9bc01c1d215b75d6c683f6ee1074a8110fd7b/compiler/crates/common/src/feature_flags.rs

And this PR (or this can be a separate PR) should just add `feature_flags` to SingleProjectConfigFile.

Reply:
I can make the change to move this to feature flags. I didn't originally since it seems like all of the feature flag config was resolved to boolean values, whereas this is a string configuration value, which didn't seem to me like a feature flag.

Reply:
> this is a string configuration value, which didn't seem to me like a feature flag.

Fair point. 

I general, I think this `field` is similar with `ConnectionInterface`. I think both of these should be defined on the `project_config` here:

https://github.com/facebook/relay/blob/59c9bc01c1d215b75d6c683f6ee1074a8110fd7b/compiler/crates/relay-compiler/src/config.rs#L252-L276

Something like `schema_configuration` field which is a struct:

```rust
struct SchemaConfiguration {
  connection_interface: ConnectionInterface,
  node_interface_id_field: Option<StringKey>,
}
```

But this refactoring can be done independently from this PR.

Reply:
or `schema_config` for short, as we already have `typegen_config`.

Reply:
@alunyov I moved it to a FeatureFlag, just so you can see what that would be like. Let me know which approach you'd like to move forward with!

Reply:
Thanks @kesne! I importing this internally.